janusgraph-core/src/main/java/org/janusgraph/diskstorage/indexing/IndexMutation.java

```diff
@@ -33,19 +34,30 @@
 
 public class IndexMutation extends Mutation<IndexEntry,IndexEntry> {
 
+    private final KeyInformation.StoreRetriever storeRetriever;
     private final boolean isNew;
     private boolean isDeleted;
 
-    public IndexMutation(List<IndexEntry> additions, List<IndexEntry> deletions, boolean isNew, boolean isDeleted) {
+    private final Function<IndexEntry, Object> entryConversionFunction =
```

Comment:
Sorry I forgot also to mention that this function was `static`. I think it should be `private static final Function<IndexEntry, Object>`.

Reply:
I see that is uses `isCollection` method, so we can make both `entryConversionFunction` and `isCollection` to be `static`.

Reply:
My bad. I see that `isCollection` is uses `storeRetriever` which isn't static.